src/ifcwrap-csharp/IfcCSharp.i

```diff
@@ -0,0 +1,377 @@
+%include "std_array.i"
+%include "std_vector.i"
+%include "std_string.i"
+%include "std_pair.i"
+%include "exception.i"
+%include "std_shared_ptr.i"
+
+%ignore IfcGeom::NumberNativeDouble;
+%ignore ifcopenshell::geometry::Converter;
+
+// General python-specific rename rules for comparison operators.
+// Mostly to silence warnings, but might be of use some time.
+%rename(Equals) operator==;
+%rename(LessThan) operator<;
+%rename(Compare) operator();
+
+%rename("Attribute") IfcParse::attribute;
+
+%exception {
+	try {
+		$action
+	} catch(const IfcParse::IfcAttributeOutOfRangeException& e) {
+		SWIG_exception(SWIG_IndexError, e.what());
+	} catch(const IfcParse::IfcException& e) {
+		SWIG_exception(SWIG_RuntimeError, e.what());
+	} catch(const std::runtime_error& e) {
+		SWIG_exception(SWIG_RuntimeError, e.what());
+	} catch(...) {
+		SWIG_exception(SWIG_RuntimeError, "An unknown error occurred");
+	}
+}
+
+%include "../serializers/serializers_api.h"
+
+%typemap(csbase) IfcGeom::IteratorSettings::Settings "long"
+
+// Include headers for the typemaps to function. This set of includes,
+// can probably be reduced, but for now it's identical to the includes
+// of the module definition below.
+%{
+	#include "../ifcgeom/Iterator.h"
+	#include "../ifcgeom/taxonomy.h"
+#ifdef IFOPSH_WITH_OPENCASCADE
+	#include "../ifcgeom/Serialization/Serialization.h"
+	#include "../ifcgeom/kernels/opencascade/IfcGeomTree.h"
+
+	#include <BRepTools_ShapeSet.hxx>
+#endif
+
+	#include "../serializers/SvgSerializer.h"
+	#include "../serializers/WavefrontObjSerializer.h"
+	#include "../serializers/HdfSerializer.h"
+	
+#ifdef HAS_SCHEMA_2x3
+	#include "../ifcparse/Ifc2x3.h"
+#endif
+#ifdef HAS_SCHEMA_4
+	#include "../ifcparse/Ifc4.h"
+#endif
+#ifdef HAS_SCHEMA_4x1
+	#include "../ifcparse/Ifc4x1.h"
+#endif
+#ifdef HAS_SCHEMA_4x2
+	#include "../ifcparse/Ifc4x2.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_rc1
+	#include "../ifcparse/Ifc4x3_rc1.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_rc2
+	#include "../ifcparse/Ifc4x3_rc2.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_rc3
+#include "../ifcparse/Ifc4x3_rc3.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_rc4
+#include "../ifcparse/Ifc4x3_rc4.h"
+#endif
+#ifdef HAS_SCHEMA_4x3
+#include "../ifcparse/Ifc4x3.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_tc1
+#include "../ifcparse/Ifc4x3_tc1.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_add1
+#include "../ifcparse/Ifc4x3_add1.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_add2
+#include "../ifcparse/Ifc4x3_add2.h"
+#endif
+
+	#include "../ifcparse/aggregate_of_instance.h"
+	
+	#include "../ifcparse/Argument.h"
+	#include "../ifcparse/IfcBaseClass.h"
+	#include "../ifcparse/IfcFile.h"
+	#include "../ifcparse/IfcSchema.h"
+	#include "../ifcparse/utils.h"
+
+	#include "../ifcgeom/ConversionSettings.h"
+	#include "../ifcgeom/ConversionResult.h"
+
+	#include "../svgfill/src/svgfill.h"
+
+#ifdef IFOPSH_WITH_CGAL
+	#include "../ifcgeom/kernels/cgal/CgalConversionResult.h"
+#endif
+%}
+
+// Create docstrings for generated python code.
+%feature("autodoc", "1");
```

Comment:
This does not generate python code.
Either adjust the comment if needed, or remove this line